astropy/io/fits/tests/test_core.py

```diff
@@ -593,15 +594,48 @@ def test_open_nonexistent(self):
             assert os.path.exists(self.temp('foobar.fits'))
             os.remove(self.temp('foobar.fits'))
 
+    def test_open_file_handle(self):
+        # Make sure we can open a FITS file from an open file handle
+        with open(self.data('test0.fits'), 'rb') as handle:
+            with fits.open(handle) as fitsfile:
+                pass
+
+        with open(self.temp('temp.fits'), 'wb') as handle:
+            with fits.open(handle) as fitsfile:
+                pass
+
+        # Opening without explicitly specifying binary mode should fail
```

Comment:
There is a difference: see [this](https://docs.python.org/3.6/tutorial/inputoutput.html#reading-and-writing-files) from the Python documentation for more information.

I'll admit I'm not familiar with all of the use cases, but if the file to be processed has any chance of containing non-decodable binary data then it is important to open with `'b'` or else later operations will fail.